plugins-experiments/src/test/kotlin/org/gradle/kotlin/dsl/experiments/plugins/GradleKotlinDslKtlintConventionPluginTest.kt

```diff
@@ -0,0 +1,248 @@
+                }
+            """
+        )
+
+        build("ktlintMainCheck")
+    }
+
+    @Test
+    fun `allowed wildcard imports`() {
+
+        withSource(
+            """
+
+                import java.util.*
+                import org.w3c.dom.*
+
+                import org.gradle.kotlin.dsl.*
+                """
+        )
+
+        buildAndFail("ktlintMainCheck")
+
+        assertKtlintErrors(1)
+        assertKtLintError("Wildcard import not allowed (org.w3c.dom.*)", 4, 17)
+    }
+
+    @Test
+    fun `blank lines`() {
+
+        withSource(
+            """
+                package some
+
+                import org.gradle.kotlin.dsl.*
+
+                val foo = "bar"
+
+                interface Foo
+
+
+
+                object Bar
+
+
+                data class Some(val name: String)
+                """
+        )
+
+        buildAndFail("ktlintMainCheck")
+
+        assertKtlintErrors(3)
+        assertKtLintError("Top level elements must be separated by two blank lines", 4, 47)
+        assertKtLintError("Top level elements must be separated by two blank lines", 6, 32)
+        assertKtLintError("Needless blank line(s)", 10, 1)
+
+        withSource(
+            """
+                package some
+
+                import org.gradle.kotlin.dsl.*
+
+
+                val foo = "bar"
+
+
+                interface Foo
+
+
+                object Bar
+
+
+                data class Some(val name: String)
+                """
+        )
+
+        build("ktlintMainCheck")
+    }
+
+    @Test
+    fun `new lines starting with ANDAND are allowed`() {
+
+        withSource(
+            """
+
+                val foo = "bar".isNotEmpty()
+                    && "bazar".isNotEmpty() // either
+            """
+        )
+
+        build("ktlintMainCheck")
+    }
+
+    @Test
+    fun `property accessors on new line`() {
+
+        withSource(
+            """
+
+            val foo get() = "bar"
+
+
+            val bar: String get() { return "bar" }
+            """
+        )
+
+        buildAndFail("ktlintMainCheck")
+
+        assertKtlintErrors(2)
+        assertKtLintError("Property accessor must be on a new line", 3, 21)
+        assertKtLintError("Property accessor must be on a new line", 6, 29)
+
+        withSource(
+            """
+
+            val foo
+                get() = "bar"
+
+
+            val bar: String
+                get() { return "bar" }
+            """
+        )
+
+        build("ktlintMainCheck")
+    }
+
+
+    private
+    fun withSource(text: String) =
+        withFile("src/main/kotlin/source.kt", text)
+
+    private
+    val ktlintReportFile by lazy { existing("build/reports/ktlint/ktlint-main.txt") }
+
+    private
+    fun assertKtlintErrors(count: Int) =
+        assertThat(
+            "ktlint error count",
+            ktlintReportFile.readLines().filter { it.contains("source.kt:") }.count(),
+            equalTo(
+                count
+            )
```

Comment:
:x: This is taking to much vertical space and in general, I think that just as in Swift, Kotlin _braces want to breathe but parentheses want to hug their contents._

Reply:
Right, the formatting in this file was not good, now fixed.